tests/testutils/kubeutils/kind_cluster.go

```diff
@@ -152,6 +162,54 @@ func (k KindCluster) Apply(manifests string) (stdOut, stdErr bytes.Buffer, err e
 	return k.runKubectl(stdin, "apply", "-f", f.Name())
 }
 
+func (k KindCluster) WaitForPods(podNameRegex, namespaceName string, duration time.Duration) []*corev1.Pod {
```

Comment:
To differentiate from pod being Ready
```suggestion
func (k KindCluster) WaitForPodsRunning(podNameRegex, namespaceName string, duration time.Duration) []*corev1.Pod {
```

 
I see us needing to wait for pods to be ready for traffic vs just in Running phase. I think at least for the collector we should wait for pod Ready if we have healthcheck extension + probes set up. In case you want to add the ready check in this PR, adding the logic (haven't tested it). 

```
import api "k8s.io/kubernetes/pkg/apis/core"

func isPodReady(pod *api.Pod) bool {
	status := pod.Status
	if status == nil {
		return false
	}
	for i := range status.Conditions {
		if status.Conditions[i].Type == api.PodReady {
			return status.Conditions[i].Status == api.ConditionTrue
		}
	}
	return false
}
```

Reply:
what about breaking out a `WaitForPodFn(podNameRegex, namespaceName string, duration time.Duration, func(*corev1.Pod) bool)` and updating `WaitForPods` to use the ready status?

Reply:
The issue with using ready for collectors in discovery is that their service extensions have yet to be running

Reply:
Ah, that makes sense